code/__HELPERS/unique_ids.dm

```diff
@@ -16,22 +16,26 @@
 
 var/global/next_unique_datum_id = 1
 
-/client/var/tmp/unique_datum_id = null
-
-/datum/proc/UID()
-	if(!unique_datum_id)
+// /client/var/tmp/unique_datum_id = null
+
+/datum/proc/UID(input)
```

Comment:
I wasn't suggesting making `UID` function like `REF`. That'll probably break things in a horrifying manner.

I was suggesting making whatever TG has work with `UID` and not `REF`.

Reply:
yeah thats kinda what I did, when theres no argument passed as input it just uses UID like it does normally but if there is an argument passed to UID it'll treat it like TG treats REF